### PR TITLE
Block middleware: Post Loop Field

### PR DESCRIPTION
**Pull request by defunctl**
## What does this do/fix?

- Replaces the Tribe ACF Post List Plugin with a Post Loop Field Middleware

## Known Limitations

- Right now you can only have one post loop field per block, if we think we'll need multiple, I should adjust this functionality before we merge.

## QA

See screencasts in the main [PR](https://github.com/moderntribe/square-one/pull/1014).

## Tests

Does this have tests?

- [x] Yes
- [ ] No, this doesn't need tests because...
- [ ] No, I need help figuring out how to write the tests.

### CODE REVIEW

**Review by dpellenwood — APPROVED** [review]
I love this, Justin! Thank you for tackling this beast. A couple thoughts:

1. I think it would be good to get design/strategy input on the field UX as they might have some suggestions for making it easier to view. Similar to how we did some field type changes and style hacking with the plugin version, we could do similar here. I'll make mention this to Sarah and see if she has space to do a review.
1. Very much thinking aloud: I wonder if we could package this up somehow to make it public as an alternative to the WP Core Query block. I realize this is probably a step backwards for our own blocks system as it benefits from the Middleware system as a whole. But, it's such a powerful tool for handling custom queries as a user.

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+			'choices' => $this->get_query_type_choices(),
+		] );
+	}
+
+	/**
+	 * Dynamic query acf field group.
+	 */
+	protected function get_query_group(): Has_Sub_Fields {
+		$group = new Field_Group( $this->config->block_name . '_' . self::GROUP_QUERY, [
+			'label'             => esc_html__( 'Build your query', 'tribe' ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php:150]:
```suggestion
			'label'             => esc_html__( 'Build Your Query', 'tribe' ),
```

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+			],
+			'default_value' => self::OPTION_DESC,
+			'multiple'      => false,
+			'ui'            => false,
+			'ajax'          => false,
+			'return_format' => 'value',
+		] );
+
+		$fields[] = new Field( $this->config->block_name . '_' . self::QUERY_ORDER_BY, [
+			'label'         => esc_html__( 'Order by', 'tribe' ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php:200]:
```suggestion
			'label'         => esc_html__( 'Order By', 'tribe' ),
```

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+			'name'          => self::QUERY_ORDER_BY,
+			'type'          => 'select',
+			'choices'       => [
+				self::OPTION_DATE  => esc_html__( 'Post date', 'tribe' ),
+				self::OPTION_TITLE => esc_html__( 'Post title', 'tribe' ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php:205]:
```suggestion
				self::OPTION_DATE  => esc_html__( 'Post Date', 'tribe' ),
				self::OPTION_TITLE => esc_html__( 'Post Title', 'tribe' ),
```

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+	 */
+	protected function build_taxonomy_fields(): array {
+		$fields     = [];
+		$taxonomies = acf_get_taxonomy_labels( $this->config->taxonomies );
+
+		asort( $taxonomies );
+
+		foreach ( $taxonomies as $taxonomy => $label ) {
+			$fields[] = new Field( $this->config->block_name . '_' . $taxonomy, [
+				'label'         => sprintf( esc_html__( 'Filter by %s', 'tribe' ), $label ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php:246]:
```suggestion
				'label'         => sprintf( esc_html__( 'Filter By %s', 'tribe' ), $label ),
```

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+					'value'    => '1',
+				],
+			],
+		] );
+
+		$fields[] = new Field( $this->config->block_name . '_' . self::MANUAL_POST_DATE, [
+			'label'             => esc_html__( 'Post Date', 'tribe' ),
+			'name'              => self::MANUAL_POST_DATE,
+			'type'              => 'date_time_picker',
+			'instructions'      => esc_html__( 'Set or override the post date', 'tribe' ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php:316]:
```suggestion
			'instructions'      => esc_html__( 'Set or override the post date.', 'tribe' ),
```

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+					'value'    => '1',
+				],
+			],
+		] );
+
+		$fields[] = new Field( $this->config->block_name . '_' . self::MANUAL_POST_AUTHOR, [
+			'label'             => esc_html__( 'Post Author', 'tribe' ),
+			'name'              => self::MANUAL_POST_AUTHOR,
+			'type'              => 'user',
+			'instructions'      => esc_html__( 'Set or override the post author', 'tribe' ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php:333]:
```suggestion
			'instructions'      => esc_html__( 'Set or override the post author.', 'tribe' ),
```

File: wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php [hunk 7]
```diff
@@ -0,0 +1,447 @@
...
+					'value'    => '1',
+				],
+			],
+		] );
+
+		$fields[] = new Field( $this->config->block_name . '_' . self::MANUAL_POST_CATEGORY, [
+			'label'             => esc_html__( 'Post Category', 'tribe' ),
+			'name'              => self::MANUAL_POST_CATEGORY,
+			'type'              => 'taxonomy',
+			'instructions'      => esc_html__( 'Set or override the primary post category', 'tribe' ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Middleware/Post_Loop/Field_Middleware/Post_Loop_Field_Middleware.php:352]:
```suggestion
			'instructions'      => esc_html__( 'Set or override the primary post category.', 'tribe' ),
```

File: wp-content/plugins/core/src/Blocks/Types/Card_Grid/Card_Grid.php [hunk 8]
```diff
@@ -80,53 +78,32 @@ public function add_fields(): void {
...
-					'limit_min'         => 2,
-					'limit_max'         => 10,
-				] )
-			);
+				'label'   => esc_html__( 'Leadin', 'tribe' ),
+				'name'    => self::LEADIN,
+				'type'    => 'text',
+				'wrapper' => [
+					'class' => 'tribe-acf-hide-label',
+				],
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Types/Card_Grid/Card_Grid.php:86]: Remove hidden labels.
```suggestion
```

File: wp-content/plugins/core/src/Blocks/Types/Card_Grid/Card_Grid.php [hunk 9]
```diff
@@ -80,53 +78,32 @@ public function add_fields(): void {
...
-					'taxonomies'        => [
-						Post_Tag::NAME,
-						Category::NAME,
-						Example::NAME,
-					],
-					'limit_min'         => 2,
-					'limit_max'         => 10,
-				] )
-			);
+				'label'   => esc_html__( 'Leadin', 'tribe' ),
```
Comment (ChrisMKindred) [inline wp-content/plugins/core/src/Blocks/Types/Card_Grid/Card_Grid.php:81]:
```suggestion
				'label'   => esc_html__( 'Overline', 'tribe' ),
```

**Review by ChrisMKindred — CHANGES_REQUESTED** [review]
Great work @defunctl.  The only changes I have are to align the field labels with the updates with #1011

**Pull request merged**